shared/src/commonMain/kotlin/com/mbta/tid/mbta_app/model/NearbyStaticData.kt

```diff
@@ -546,8 +564,21 @@ fun NearbyStaticData.withRealtimeInfo(
         }
         .filterNot { it.isEmpty() }
         .toList()
-        .sortedWith(compareBy({ it.distanceFrom(sortByDistanceFrom) }, { it.sortRoute() }))
-        .sortedWith(compareBy(Route.relevanceComparator(pinnedRoutes)) { it.sortRoute() })
+        .sortedWith(
+            compareBy<StopsAssociated, Route>(Route.pinnedRoutesComparator(pinnedRoutes)) {
+                    it.sortRoute()
+                }
+                .thenBy {
+                    it.patternsByStop.all { byStop ->
+                        byStop.patterns.all { patterns ->
+                            patterns.upcomingTrips.isNullOrEmpty() && !patterns.hasSchedulesToday
+                        }
+                    }
+                }
+                .thenBy(Route.subwayFirstComparator) { it.sortRoute() }
+                .thenBy { it.distanceFrom(sortByDistanceFrom) }
+                .thenBy { it.sortRoute() }
+        )
```

Comment:
I got stuck on the sorting for a bit, until I realized that I needed to split up the `Route.relevanceComparator`, because we needed to sort the non-scheduled routes after the pinning but before putting subway first. This is risky in case we add more steps to the `relevanceComparator`, it would be awkward to get the schedule sorting to conditionally happen within there.